app/models/manageiq/providers/openshift/container_manager/refresh_parser.rb

```diff
@@ -10,9 +10,34 @@ def ems_inv_to_hashes(inventory, options = Config::Options.new)
         get_templates(inventory)
         get_or_merge_openshift_images(inventory) if options.get_container_images
         EmsRefresh.log_inv_debug_trace(@data, "data:")
+
+        # Returning a hash triggers save_inventory_container code path.
         @data
       end
 
+      def initialize_inventory_collections(ems, options)
+        super
+        # get_container_images=false mode is a stopgap to speed up refresh by reducing functionality.  When it's flipped from true to false, we should at least retain existing metadata.
+        if options.get_container_images
+          initialize_custom_attributes_collections(ems.container_images, %w(labels docker_labels))
+        end
+      end
+
+      # We rely on Kubernetes initialize_inventory_collections setting up our collections too.
+      # TODO: does it mean kubernetes refresh would wipe out the openshift tables?
```

Comment:
Yeah, if you would run kubernetes refresh on openshift provider, it would delete the tables you are not sending data for. But we don't do that (running another's manager type refresh), since that would break anywhere. :-)